File: backend/art-galery/src/main/java/com/unmsm/movil/tecnisis/art_galery/infrastructure/adapters/input/rest/model/response/ArtWorkResponse.java
package com.unmsm.movil.tecnisis.art_galery.infrastructure.adapters.input.rest.model.response;

import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@Builder @Getter @Setter
@NoArgsConstructor @AllArgsConstructor
public class ArtWorkResponse {
    private Long id;
    private String title;
    private LocalDate creationDate;
    private String image;
    private BigDecimal height;
    private BigDecimal width;
    private ArtistResponse artist;
    private TechniqueResponse technique;
}

File: tecnisis-app/app/src/main/java/com/example/tecnisis/data/artwork/ArtworkResponse.kt
package com.example.tecnisis.data.artwork

import com.example.tecnisis.data.artist.ArtistResponse
import com.example.tecnisis.data.technique.TechniqueResponse
import com.google.gson.annotations.SerializedName

data class ArtworkResponse(
    val id: Long,
    val title: String,
    @SerializedName("creation_date")
    val creationDate: String,
    val image: String,
    val width: Double,
    val height: Double,
    val technique: TechniqueResponse,
    val artist: ArtistResponse
)
